api_app/service_bus/deployment_status_update.py

```diff
@@ -18,217 +22,227 @@
-                await send_deployment_message(content=content, correlation_id=operation.id, session_id=resource_to_send.id, action=next_step.resourceAction)
             except Exception as e:
-                logging.error(f"Unable to send update for resource in pipeline step: {e}")
-                next_step.message = repr(e)
-                next_step.status = Status.UpdatingFailed
-                update_overall_operation_status(operation, next_step, is_last_step, resource_repo)
-                operations_repo.update_item(operation)
-
-        result = True
-
-    except EntityDoesNotExist:
-        # Marking as true as this message will never succeed anyways and should be removed from the queue.
-        result = True
-        error_string = strings.DEPLOYMENT_STATUS_ID_NOT_FOUND.format(message.id)
-        logging.error(error_string)
-    except Exception as e:
-        logging.error(strings.STATE_STORE_ENDPOINT_NOT_RESPONDING + " " + str(e))
-
-    return result
-
-
-async def receive_message_and_update_deployment(app) -> None:
-    """
-    Receives messages from the deployment status update queue and updates the status for
-    the associated resource in the state store.
-    Args:
-        app ([FastAPI]): Handle to the currently running app
-    """
-    receive_message_gen = receive_message()
-
-    try:
-        async for message in receive_message_gen:
-            operations_repo = OperationRepository(get_db_client(app))
-            resource_repo = ResourceRepository(get_db_client(app))
-            resource_template_repo = ResourceTemplateRepository(get_db_client(app))
-            result = await update_status_in_database(resource_repo, operations_repo, resource_template_repo, message)
-            await receive_message_gen.asend(result)
-    except StopAsyncIteration:  # the async generator when finished signals end with this exception.
-        pass
+                # Catch all other exceptions, log them via .exception to get the stack trace, and reconnect
+                logging.exception(f"Unknown exception. Will retry - {e}")
+
+    async def process_message(self, msg):
+        complete_message = False
+        message = ""
+
+        try:
+            message = parse_obj_as(DeploymentStatusUpdateMessage, json.loads(str(msg)))
+            logging.info(f"Received and parsed JSON for: {msg.correlation_id}")
+            complete_message = await self.update_status_in_database(message)
+            logging.info(f"Update status in DB for {message.operationId} - {message.status}")
+        except (json.JSONDecodeError, ValidationError) as e:
+            complete_message = True
+            logging.error(f"{strings.DEPLOYMENT_STATUS_MESSAGE_FORMAT_INCORRECT}: {msg.correlation_id} - {e}")
+        except Exception as e:
+            logging.info(f"Exception for: {msg.correlation_id} - {e}")
+
+        return complete_message
+
+    async def update_status_in_database(self, message: DeploymentStatusUpdateMessage):
```

Comment:
From here on the code hasn't changed, right?

Reply:
correct, the `update_status_in_database` does the same thing - the only stuff that's changed is that now we're using a class we have some class level members (the repositories) that we set up once at startup, so we use `self.repo` instead of passing `repo` around